### PR TITLE
create repository : remote

### PR DESCRIPTION
**Pull request by trantan97**
## [Pull request title]
create repository : remote
### [Redmine link]
https://edu-redmine.framgia.vn/issues/7944

### CODE REVIEW

**Comment by trantan97** [comment]
ready

File: app/build.gradle [hunk 1]
```diff
@@ -31,12 +31,17 @@ dependencies {
...
-    implementation 'io.reactivex.rxjava2:rxandroid:2.1.0'
-    implementation 'com.squareup.okhttp3:okhttp:3.12.0'
+    implementation 'com.squareup.retrofit2:adapter-rxjava2:2.4.0'
+    //Glide
+    implementation 'com.github.bumptech.glide:glide:4.8.0'
+    annotationProcessor 'com.github.bumptech.glide:compiler:4.8.0'
+    //RxJava and RxAndroid
+    implementation 'io.reactivex.rxjava2:rxandroid:2.1.1'
+    implementation 'io.reactivex.rxjava2:rxjava:2.2.7'
+    //
```
Comment (NguyenMinhTuanB) [inline app/build.gradle]: remove this line

File: app/src/main/java/com/ptit/filmdictionary/data/source/remote/RemoteDataSource.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import android.content.Context;
+
+import com.ptit.filmdictionary.data.model.Movie;
+import com.ptit.filmdictionary.data.source.MovieDataSource;
+import com.ptit.filmdictionary.data.source.remote.response.GenreResponse;
+import com.ptit.filmdictionary.data.source.remote.response.MovieResponse;
+
+import io.reactivex.Observable;
+
+public class RemoteDataSource implements MovieDataSource.Remote {
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/ptit/filmdictionary/data/source/remote/RemoteDataSource.java]:
```suggestion
public class MovieRemoteDataSource implements MovieDataSource.Remote {
```

File: app/src/main/java/com/ptit/filmdictionary/data/source/remote/ApiRequest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+public interface ApiRequest {
+    @GET("genre/movie/list")
+    Observable<GenreResponse> getGenres();
+
+    @GET("{type}")
+    Observable<MovieResponse> getMovieByCategory(@Path("type") String type,
+                                                 @Query("page") int page);
+
+    @GET("discover/movie")
+    Observable<MovieResponse> getMovieByGenre(@Query("with_genres") String idGenre,
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/ptit/filmdictionary/data/source/remote/ApiRequest.java]:
```suggestion
    Observable<MovieResponse> getMoviesByGenre(@Query("with_genres") String idGenre,
```

File: app/src/main/java/com/ptit/filmdictionary/data/source/remote/ApiRequest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+public interface ApiRequest {
+    @GET("genre/movie/list")
+    Observable<GenreResponse> getGenres();
+
+    @GET("{type}")
+    Observable<MovieResponse> getMovieByCategory(@Path("type") String type,
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/com/ptit/filmdictionary/data/source/remote/ApiRequest.java]:
```suggestion
    Observable<MovieResponse> getMoviesByCategory(@Path("type") String type,
```

**Comment by trantan97** [comment]
ready

**Comment by trantan97** [comment]
ready

**Pull request merged**